podaac/subsetter/dimension_cleanup.py

```diff
@@ -28,46 +28,69 @@ def remove_duplicate_dims(nc_dataset: nc.Dataset) -> Tuple[nc.Dataset, List[str]
-        data[dim_dup_new] = nc_dataset.createVariable(dim_dup_new, nc_dataset.variables[dim_dup].dtype,
-                                                      (dim_dup_new,), fill_value=fill_value)
 
-        for ncattr in nc_dataset.variables[dim_dup].ncattrs():
-            if ncattr != '_FillValue':
-                data[dim_dup_new].setncattr(ncattr, nc_dataset.variables[dim_dup].getncattr(ncattr))
-        data[dim_dup_new][:] = nc_dataset.variables[dim_dup][:]
+        # Only create a new *Dimension* if it doesn't already exist.
+        if dim_dup_new not in nc_dataset.dimensions.keys():
 
-        new_dim_list = dim_list[:-1]
-        new_dim_list.extend([dim_dup_new])
+            # New dimension is created by copying from the duplicated dimension.
+            nc_dataset.createDimension(dim_dup_new, dim_dup_length)
 
-        # createVariable with new dimensions
+            # Only create a new dimension *Variable* if it existed originally in the NetCDF structure.
+            if dim_dup in nc_dataset.variables.keys():
 
-        data[var_name_new] = nc_dataset.createVariable(var_name_new, str(dup_var[:].dtype), tuple(new_dim_list), fill_value=fill_value)
+                # New variable object is created for the renamed, previously duplicated dimension.
+                new_dup_var[dim_dup_new] = nc_dataset.createVariable(dim_dup_new, nc_dataset.variables[dim_dup].dtype,
+                                                                     (dim_dup_new,), fill_value=fill_value)
+                # New variable's attributes are set to the original ones.
+                for ncattr in nc_dataset.variables[dim_dup].ncattrs():
+                    if ncattr != '_FillValue':
+                        new_dup_var[dim_dup_new].setncattr(ncattr, nc_dataset.variables[dim_dup].getncattr(ncattr))
+                new_dup_var[dim_dup_new][:] = nc_dataset.variables[dim_dup][:]
 
-        for attrname in dup_var.ncattrs():
-            if attrname != '_FillValue':
-                data[var_name_new].setncattr(attrname, nc_dataset.variables[dup_var_name].getncattr(attrname))
-                data[var_name_new][:] = nc_dataset.variables[dup_var_name][:]
+        # Delete existing Variable
         del nc_dataset.variables[dup_var_name]
 
+        # Replace original *Variable* with new variable with no duplicated dimensions.
+        new_dup_var[dup_var_name] = nc_dataset.createVariable(dup_var_name, str(dup_var[:].dtype),
+                                                              tuple(new_dim_list), fill_value=fill_value)
+        for attr_name, contents in attrs_contents.items():
+            new_dup_var[dup_var_name].setncattr(attr_name, contents)
+        new_dup_var[dup_var_name][:] = dup_var[:]
+
     # return the variables that will need to be renamed: Rename method is still an issue per https://github.com/Unidata/netcdf-c/issues/1672
-    return nc_dataset, dup_new_varnames
+    return nc_dataset, []
```

Comment:
If the new variable names with duplicate dimensions are the same as the original name (without the '_1' at the end) then a blank list doesn't need to be returned and the function below should be removed. Will need to check with our products.

Reply:
Yep! I've updated the branch now to remove that return value and the use of the other renaming function.